### PR TITLE
Refactor BT automation for private DC use case

### PR DESCRIPTION
**Pull request by shifucun**
Extract common logic into common.go.

Have separate "main" file for prod and private deployment, where the logic differs by the folder structure.

The private import assumes the /control and /cache folder are sibling folders under one import group folder.

### CODE REVIEW

File: bigtable_automation/gcf/prod/base.yaml [hunk 1]
```diff
@@ -2,7 +2,7 @@ projectID: "datcom-store"
 instance: "prophet-cache"
 cluster: "prophet-cache-c1"
 nodesHigh: "298"
-nodesLow: "20"
+nodesLow: "30"
```
Comment (shifucun) [inline bigtable_automation/gcf/prod/base.yaml:5]: @Spaceenter so we want to change this one?

File: bigtable_automation/gcf/private.go [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package private runs a GCF function that triggers in 2 scenarios:
+//
+// 1) completion of prophet-flume job in borg. On triggering it sets up new
+//    cloud BT table, scales up BT cluster (if needed) and starts a dataflow job.
+// 2) completion of BT cache ingestion dataflow job. It scales BT cluster down
+//    (if needed).
+// The trigger function BTImportController requires a set of environment
```
Comment (pradh) [inline bigtable_automation/gcf/private.go:21]: fix name

File: bigtable_automation/gcf/prod.go [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package gcf runs a GCF function that triggers in 2 scenarios:
+//
+// 1) completion of prophet-flume job in borg. On triggering it sets up new
+//    cloud BT table, scales up BT cluster (if needed) and starts a dataflow job.
+// 2) completion of BT cache ingestion dataflow job. It scales BT cluster down
+//    (if needed).
+// The trigger function BTImportController requires a set of environment
```
Comment (pradh) [inline bigtable_automation/gcf/prod.go]: fix name

File: bigtable_automation/gcf/private.go [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package private runs a GCF function that triggers in 2 scenarios:
+//
+// 1) completion of prophet-flume job in borg. On triggering it sets up new
+//    cloud BT table, scales up BT cluster (if needed) and starts a dataflow job.
+// 2) completion of BT cache ingestion dataflow job. It scales BT cluster down
+//    (if needed).
```
Comment (pradh) [inline bigtable_automation/gcf/private.go]: What is different wrt prod?

File: bigtable_automation/gcf/private.go (hunk 2 above)
Comment (shifucun) [reply bigtable_automation/gcf/private.go]: Removed these since the package comment is in common.go

File: bigtable_automation/gcf/private.go [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+		return errors.Wrap(err, "Unable to parse 'nodesHigh' as an integer")
+	}
+	nodesL, err := strconv.Atoi(nodesLow)
+	if err != nil {
+		return errors.Wrap(err, "Unable to parse 'nodesLow' as an integer")
+	}
+	// Get table ID.
+	// e.Name should is like "**/<user>/<import>/control/<table_id>/launched.txt"
+	parts := strings.Split(e.Name, "/")
+	if parts[len(parts)-3] != "control" {
```
Comment (pradh) [inline bigtable_automation/gcf/private.go:59]: might be more readable to use tmp vars?
idxControl := len(parts) - 3
idxTable := len(parts) -2

**Review by pradh — COMMENTED** [review]
Also update about private in:
https://github.com/datacommonsorg/tools/tree/master/bigtable_automation#bigtable-automation
https://github.com/datacommonsorg/tools/blob/master/bigtable_automation/gcf/README.md#bigtable-automation
?

File: bigtable_automation/gcf/common.go [hunk 6]
```diff
@@ -206,123 +168,38 @@ func getBTNodes(ctx context.Context, projectID, instance, cluster string) (int,
...
-	}
-	tableID := parts[len(parts)-2]
-	triggerFolder := strings.Join(parts[0:len(parts)-2], "/")
-	if triggerFolder != controlFolder {
-		log.Printf("Control folder '%s' != '%s', skip processing", triggerFolder, controlFolder)
-		return nil
-	}
-
-	numNodes, err := getBTNodes(ctx, projectID, instance, cluster)
+func launchDataflowJob(
```
[the remark this answers is not in the document]
Comment (shifucun) [reply bigtable_automation/gcf/common.go]: Done.

**Comment by shifucun** [comment]
> Also update about private in: https://github.com/datacommonsorg/tools/tree/master/bigtable_automation#bigtable-automation https://github.com/datacommonsorg/tools/blob/master/bigtable_automation/gcf/README.md#bigtable-automation ?

Done

File: bigtable_automation/README.md [hunk 7]
```diff
@@ -2,32 +2,38 @@
...
+   per-version state files located in GCS:
 
-    *   Base cache:
-        [/bigstore/datcom-control/base/{BT_Table}/...](https://pantheon.corp.google.com/storage/browser/datcom-control/base)
-    *   Branch cache:
-        [/bigstore/datcom-control/branch/{BT_Table}/...](https://pantheon.corp.google.com/storage/browser/datcom-control/branch)
+   - Base cache:
+     [/bigstore/datcom-control/base/{BT_Table}/...](https://pantheon.corp.google.com/storage/browser/datcom-control/base)
+   - Branch cache:
+     [/bigstore/datcom-control/branch/{BT_Table}/...](https://pantheon.corp.google.com/storage/browser/datcom-control/branch)
```
Comment (pradh) [inline bigtable_automation/README.md:23]: Don't the paths change across private vs. prod?

File: bigtable_automation/gcf/prod/branch.yaml [hunk 8]
```diff
@@ -1,8 +1,8 @@
 projectID: "datcom-store"
 instance: "prophet-branch-cache"
 cluster: "prophet-branch-cache-c1"
-nodesHigh: "3"
-nodesLow: "1"
+nodesHigh: "30"
+nodesLow: "3"
```
Comment (pradh) [inline bigtable_automation/gcf/prod/branch.yaml:5]: I'm not sure that this need to be 30?  5?

File: bigtable_automation/gcf/prod/base.yaml (hunk 1 above)
Comment (pradh) [reply bigtable_automation/gcf/prod/base.yaml:5]: Yes

**Review by pradh — APPROVED** [review]

File: bigtable_automation/gcf/prod/branch.yaml (hunk 8 above)
Comment (shifucun) [reply bigtable_automation/gcf/prod/branch.yaml]: Done

File: bigtable_automation/README.md (hunk 7 above)
Comment (shifucun) [reply bigtable_automation/README.md:23]: Added path for private

File: bigtable_automation/gcf/prod/base.yaml (hunk 1 above)
Comment (Spaceenter) [reply bigtable_automation/gcf/prod/base.yaml:5]: yes

**Review by Spaceenter — DISMISSED** [review]

**Review by pradh — APPROVED** [review]

**Comment by shifucun** [comment]
thanks!

**Pull request merged**